lib/sql_footprint.rb

```diff
@@ -1,88 +1,64 @@
 require 'sql_footprint/version'
+require 'set'
+require 'active_support/notifications'
 
 module SqlFootprint
-  def self.start
-    @original_logger = ActiveRecord::Base.logger
-    @logger = Logger.new
-    ActiveRecord::Base.logger = @logger
+  ActiveSupport::Notifications.subscribe('sql.active_record') do |_, _, _, _, payload|
+    capture payload.fetch(:sql)
   end
 
-  def self.stop
-    ActiveRecord::Base.logger = @original_logger
-    File.open('footprint.sql', 'w') do |f|
-      @logger.logs.each do |log|
-        f.puts log
-      end
-    end
-  end
-
-  def self.exclude
-    ActiveRecord::Base.logger = @original_logger
-    yield
-  ensure
-    ActiveRecord::Base.logger = @logger
-  end
-
-  class Logger
-    def initialize
-      @logs = []
+  class << self
+    def start
+      @capture = true
+      @lines   = Set.new
     end
 
-    attr_reader :logs
-
-    def error param
-      raise param
-    end
-
-    def debug text
-      if sql? text
-        sql = format_sql(text)
-        logs << sql unless logs.include?(sql)
-        logs.sort!
+    def stop
+      @capture = false
+      File.open('footprint.sql', 'w') do |f|
+        lines.each do |line|
+          f.puts line
+        end
       end
     end
 
-    def debug?
-      true
+    def exclude
+      @capture = false
```

Comment:
Is it common practice to use an instance variable named the same as the method it's toggling on and off? It threw me off initially that we have an instance variable named `@capture` and a method named `capture` that looks like a setter.